crates/gitbutler-project-store/src/migrations/migrator.rs

```diff
@@ -0,0 +1,204 @@
+use anyhow::{Context, Ok, Result};
+use rusqlite::{Connection, TransactionBehavior};
+use uuid::Uuid;
+
+use super::migration::Migration;
+
+/// SQL required to create the migrations table.
+///
+/// This should only ever be changed with great caution
+const CREATE_MIGRATIONS_TABLE: &str = "
+CREATE TABLE IF NOT EXISTS migrations (
+    id BLOB PRIMARY KEY NOT NULL, -- A V4 UUID
+    name TEXT NOT NULL
+)
+";
+
+/// Migrator is used to perform migrations on a particular database.
+pub(crate) struct Migrator<'l> {
+    connection: &'l mut Connection,
+}
+
+impl<'l> Migrator<'l> {
+    pub(crate) fn new(connection: &'l mut Connection) -> Migrator<'l> {
+        Migrator { connection }
+    }
+
+    /// Iterates over the list of provided migrations starting at index 0.
+    /// If the migration has already been run, it will be skipped.
+    /// Run migrations get recorded in the `migrations` table.
+    /// The `migrations` table will be created if it doesn't exist.
+    pub(crate) fn migrate(&mut self, migrations: Vec<Migration>) -> Result<()> {
```

Comment:
`migrations` doesn't need to be consumed, and when I read it I wonder why that needs to be consumed. Better not 'lie' and do `&[Migration]` instead.